File: 프로그래머스/0/181901. 배열 만들기 1/배열 만들기 1.java
import java.util.stream.*;

class Solution {
    public int[] solution(int n, int k) {
        
        return IntStream.rangeClosed(1, n)
                 .filter(num -> num % k == 0)
                 .toArray();
    }
}

// 다른 풀이
int count = n / k;
int[] answer = new int[count];

for (int i = 1; i <= count; i++) {
    answer[i - 1] = k * i;
}

return answer;

File: 프로그래머스/0/181901. 배열 만들기 1/배열 만들기 1.py
def solution(n, k):
    
    return [num for num in range(1, n + 1) if num % k == 0]

# 다른 풀이
return [i for i in range(k, n+1, k)] # k 부터 n+1까지 k의 배수로
